Realm/Realm/Linq/WhereClauseVisitor.cs

```diff
@@ -0,0 +1,329 @@
+
+                internal static readonly LazyMethod EqualsStringComparison = Capture<string>(s => s.Equals(string.Empty, StringComparison.Ordinal));
+            }
+        }
+
+        private readonly RealmObjectBase.Metadata _metadata;
+
+        private WhereClause _whereClause;
+
+        public WhereClauseVisitor(RealmObjectBase.Metadata metadata)
+        {
+            _metadata = metadata;
+            _whereClause = new WhereClause();
+        }
+
+        public WhereClause VisitWhere(LambdaExpression whereClause)
+        {
+            _whereClause.ExpNode = Extract(whereClause.Body);
+            var json = JsonConvert.SerializeObject(_whereClause, formatting: Formatting.Indented);
+            Visit(whereClause.Body);
+            return _whereClause;
+        }
+
+        private ExpressionNode Extract(Expression node)
+        {
+            var realmLinqExpression = Visit(node) as RealmLinqExpression;
+            return realmLinqExpression.ExpressionNode;
+        }
+
+        protected override Expression VisitBinary(BinaryExpression be)
+        {
+            ExpressionNode returnNode;
+            if (be.NodeType == ExpressionType.AndAlso)
+            {
+                var andNode = new AndNode();
+                andNode.Left = Extract(be.Left);
+                andNode.Right = Extract(be.Right);
+                returnNode = andNode;
+            }
+            else if (be.NodeType == ExpressionType.OrElse)
+            {
+                var orNode = new OrNode();
+                orNode.Left = Extract(be.Left);
+                orNode.Right = Extract(be.Right);
+                returnNode = orNode;
+            }
+            else
+            {
+                ComparisonNode comparisonNode;
+                switch (be.NodeType)
+                {
+                    case ExpressionType.Equal:
+                        comparisonNode = new EqualityNode();
+                        break;
+                    case ExpressionType.NotEqual:
+                        comparisonNode = new NotEqualNode();
+                        break;
+                    case ExpressionType.LessThan:
+                        comparisonNode = new LtNode();
+                        break;
+                    case ExpressionType.LessThanOrEqual:
+                        comparisonNode = new LteNode();
+                        break;
+                    case ExpressionType.GreaterThan:
+                        comparisonNode = new GtNode();
+                        break;
+                    case ExpressionType.GreaterThanOrEqual:
+                        comparisonNode = new GteNode();
+                        break;
+                    default:
+                        throw new NotSupportedException($"The binary operator '{be.NodeType}' is not supported");
+                }
+
+                if (be.Left is MemberExpression me)
```

Comment:
Here we are considering that in the expression the property is on the left and the value is on the right, like "p.Name == "alice". What would happen if they're inverted, for example "alice == p.Name"? 
This is also something that needs to be discussed with Jakob, in case it makes a difference